socs/agent/pmx.py

```diff
@@ -38,24 +38,51 @@ def __del__(self):
             pass
         return
 
+    def check_connect(self):
+        """Check the connection."""
+        try:
+            if not self.using_tcp :
+                self.ser.inWaiting()
+            else  :
+                self.clean_serial()
+                self.wait()
+                self.ser.write("OUTP?\n\r")
+                self.wait()
+                val = (self.ser.readline().strip())
+                val = int(val)
+        except Exception as e:
+            msg = 'Could not connect to the PMX serial! | Error: "{}"'.format(e)
+            return msg, False
+        return 'Successfully connect to the PMX serial.', True
+
     def check_voltage(self):
         """Check the voltage."""
         self.clean_serial()
         bts = self.ser.write("MEAS:VOLT?\n\r")
         self.wait()
-        val = float(self.ser.readline())
-        msg = "Measured voltage = %.3f V" % (val)
-        # print(msg)
+        try :
+            val = float(self.ser.readline())
+            msg = "Measured voltage = %.3f V" % (val)
+            # print(msg)
+        except  ValueError:
+            msg = 'WARNING! Could not get correct voltage value! | Response = "%s"' % (val)
+            val = -999
+            print(msg)
         return msg, val
```

Comment:
I don't understand clearly.
There is an initialization of ``val`` in the except section.
Is it not enough?

By the way, I noticed that the ``val`` should be float type (-999.) instead of int() (-999).
So I changed several initial values to -999.

This exception can occur if the read string in ``self.ser.eadline()`` has a non-number character.
This can occur if the communication between the PC and KIKUSUI has some trouble. 
I hope this will not occur in real usage at the site. But, it may occur in the commissioning phase or lab. test.

Reply:
> I don't understand clearly. There is an initialization of `val` in the except section. Is it not enough?

Just looking at this segment here, the initialization is on line 69, but `val` is used in the string formatting on line 68, which will fail with: `NameError: name 'val' is not defined.` Swapping these lines would work.